components/core/src/clp_s/log_converter/LogConverter.cpp

```diff
@@ -0,0 +1,141 @@
+#include <log_surgeon/Constants.hpp>
+#include <log_surgeon/Schema.hpp>
+#include <ystdlib/containers/Array.hpp>
+#include <ystdlib/error_handling/Result.hpp>
+
+#include "../../clp/ErrorCode.hpp"
+#include "../../clp/ReaderInterface.hpp"
+#include "../InputConfig.hpp"
+#include "LogSerializer.hpp"
+
+namespace clp_s::log_converter {
+namespace {
+/**
+ * Non-exhaustive timestamp schema which covers many common patterns.
+ *
+ * Once log-surgeon has better unicode support, we should also allow \u2202 as an alternative
+ * minus sign for timezone offsets.
+ */
+constexpr std::string_view cTimestampSchema{
+        R"(timestamp:(\d{2,4}[ /\-]{0,1}[ 0-9]{2}[ /\-][ 0-9]{2})|([ 0-9]{2}[ /\-])"
+        R"(((Jan(uary){0,1})|(Feb(ruary){0,1})|(Mar(ch){0,1})|(Apr(il){0,1})|(May)|(Jun(e){0,1})|)"
+        R"((Jul(y){0,1})|(Aug(ust){0,1})|(Sep(tember){0,1})|(Oct(ober){0,1})|(Nov(ember){0,1})|)"
+        R"((Dec(ember){0,1}))[ /\-]\d{2,4})[ T:][ 0-9]{2}:[ 0-9]{2}:[ 0-9]{2})"
+        R"(([,\.:]\d{1,9}){0,1}([ ]{0,1}(UTC){0,1}[\+\-]\d{2}(:{0,1}\d{2}){0,1}Z{0,1}){0,1})"
+};
+constexpr std::string_view cDelimiters{R"(delimiters: \t\r\n\[\(:)"};
+}  // namespace
+
+auto LogConverter::refill_buffer(std::shared_ptr<clp::ReaderInterface>& reader)
+        -> ystdlib::error_handling::Result<size_t> {
+    compact_buffer();
+    YSTDLIB_ERROR_HANDLING_TRYV(grow_buffer_if_full());
+
+    size_t num_bytes_read{};
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    auto const rc{reader->try_read(
+            m_buffer.data() + m_bytes_occupied,
+            m_buffer.size() - m_bytes_occupied,
+            num_bytes_read
+    )};
+    // NOLINTEND(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    m_bytes_occupied += num_bytes_read;
+    if (clp::ErrorCode_EndOfFile == rc) {
+        return num_bytes_read;
+    }
+    if (clp::ErrorCode_Success != rc) {
+        return std::errc::not_enough_memory;
+    }
+
+    return num_bytes_read;
+}
+
+void LogConverter::compact_buffer() {
+    if (m_cur_offset > 0) {
+        // NOLINTBEGIN(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+        std::memmove(
+                m_buffer.data(),
+                m_buffer.data() + m_cur_offset,
+                m_bytes_occupied - m_cur_offset
+        );
+        // NOLINTEND(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+        m_bytes_occupied -= m_cur_offset;
+        m_cur_offset = 0;
+    }
+}
+
+auto LogConverter::grow_buffer_if_full() -> ystdlib::error_handling::Result<void> {
+    if (m_buffer.size() == m_bytes_occupied) {
+        size_t const new_size{2 * m_buffer.size()};
+        if (new_size > cMaxBufferSize) {
+            return std::errc::result_out_of_range;
+        }
+        ystdlib::containers::Array<char> new_buffer(new_size);
+        std::memcpy(new_buffer.data(), m_buffer.data(), m_bytes_occupied);
+        m_buffer = std::move(new_buffer);
+    }
+    return ystdlib::error_handling::success();
```

Comment:
```suggestion
    if (m_buffer.size() != m_bytes_occupied) {
        return ystdlib::error_handling::success();
    }

    size_t const new_size{2 * m_buffer.size()};
    if (new_size > cMaxBufferSize) {
        return std::errc::result_out_of_range;
    }
    ystdlib::containers::Array<char> new_buffer(new_size);
    std::memcpy(new_buffer.data(), m_buffer.data(), m_bytes_occupied);
    m_buffer = std::move(new_buffer);
    return ystdlib::error_handling::success();
```
Similarly.
btw I'd suggest to try `std::copy` and benchmark the performance (I think it should be optimized to use `std::memcpy`), but if we're running out of time I'm fine to use `std::memcpy` for now.